CreeDictionary/utils/crkeng_xml_utils.py

```diff
@@ -79,3 +106,237 @@ def convert_xml_inflectional_category_to_word_class(
         return WordClass.IPV
 
     return None
+
+
+class IndexedXML(Iterable[XMLEntry]):
+    """
+    The collection of all entries parsed from the XML source file.
+    It supports fast querying with field/field combinations by hashing the entries in advance.
+    """
```

Comment:
this is the **crux** of the PR. `Iterable[XMLEntry]` is more semantical than practical (also `__iter__` is indeed used so it's really good)